syntax = "proto3";

option go_package = "volpe-framework/comms/common";

message Individual {
  bytes genotype = 1;
  float fitness = 2;
}

message Population {
  repeated Individual members = 1;
  optional string problemID = 2;
}

message ImageStreamObject {
  oneof data {
    ImageDetails details = 1;
    ImageChunk chunk = 2;
  }
}

message ImageDetails {
  string problemID = 1;
  int32 imageSizeBytes = 2;
}

message ImageChunk {
  bytes data = 1;
}



// TODO: Add close msgs
